junorunner/testrunner.py

```diff
@@ -30,18 +37,27 @@ def add_arguments(cls, parser):
             action='store',
             dest='slow_test_count',
             default=0,
-            help="Print given number of slowest tests"
+            help='Print given number of slowest tests'
+        )
+        parser.add_argument('--only-failed',
+            action='store_true',
+            help='Run only failed tests'
         )
 
     def run_tests(self, test_labels, extra_tests=None, **kwargs):
         """
         Run the unit tests for all the test labels in the provided list.
         """
 
+        if self.only_failed:
+            with open(RERUN_LOG_FILE_NAME, 'r') as f:
+                test_labels = f.read().split('\n')
+            test_labels += [':']
```

Comment:
Why?

Reply:
because : seperates test case names and test names. Without : it will not work unfortunatelly